src/cargo/lib.rs

```diff
@@ -234,7 +234,24 @@ fn flags_from_args<'a, T>(usage: &str, args: &[String],
                                    .version(Some(version()));
     docopt.decode().map_err(|e| {
         let code = if e.fatal() {1} else {0};
-        CliError::from_error(human(e.to_string()), code)
+        match e {
+            DoError::WithProgramUsage(ref nm, ref usgm) => {
+                if code == 1 {
+                    let nflag = &nm.to_string()[15..];
```

Comment:
This seems like it may be a bit too magical parsing the error message in an infallible fashion (e.g. `unwrap()`s, etc). Perhaps docopt has a more structured form of this error that could be detected?